llvm-to-smt/generate_encodings.py

```diff
@@ -711,6 +715,14 @@ def print_and_log(s, pend="\n"):
         # only support modular verification in kernels with "reg_bounds_sync"
         assert version.parse(args.kernver) >= version.parse("5.19-rc6")
 
+    if args.bv_suffix_override:
+        if args.specific_op is None:
+            raise RuntimeError(
+            '--bv-suffix-override can only be used with --specific-op')
```

Comment:
That's why wanted to run it by you. 

I wanted to make sure we know exactly what we are doing when we use `--bv-suffix-override`. It is important that encodings use the same bitvector suffix across runs (that's why the hardcoded suffixes as you know). We would only need different suffixes for checking for equivalence, in which case we should use specific-op, so there is no need for it otherwise as such. 

We could also make `--bv-suffix-override`  accept an integer that would be added instead of the `1000` currently. I'm wondering if we would ever need that, or is it good to have anyway.

Reply:
> We would only need different suffixes for checking for equivalence, in which case we should use specific-op, so there is no need for it otherwise as such.

I think I'm missing something. Why *should* we use specific-op if we're checking for equivalence? Couldn't we compare equivalence for all ops at once?

Reply:
> We could also make `--bv-suffix-override` accept an integer that would be added instead of the `1000` currently. I'm wondering if we would ever need that, or is it good to have anyway.

I wondered the same but I think you were right to make it a simple boolean. We can always change it later if we need to.

Reply:
> > We would only need different suffixes for checking for equivalence, in which case we should use specific-op, so there is no need for it otherwise as such.
> 
> I think I'm missing something. Why _should_ we use specific-op if we're checking for equivalence? Couldn't we compare equivalence for all ops at once?

Checking for equivalence takes (especially for jumps whcih include sync) time (see comments in https://github.com/bpfverif/agni/pull/37). So I figured we would only do a few ops at a time (in the CI, or otherwise).  Checking for *all* ops will definitely take upwards of a 2-3 days at least, so I don't  think we would be able to do it in CI. 

The bitvector suffixes are very important for equivalence check.  I have spent a few days sometimes trying to debug the output of smt_equiv.py only to later realize that suffixes of the two encodings being compared were the same, so the output of smt_equiv.py was garbage.  

I was thinking if there is a way for equivalence check to enforce different bitvector suffixes. I tried this using a regex check of the smt files, but it isn't straightforward. There might be a way to have LLVMToSMT output the suffix that was used in the smt file itself (as comments); then smt_equiv.py can simply confirm if the suffixes are different.

Reply:
> Checking for equivalence takes time (especially for jumps whcih include sync) (see comments in #37). So I figured we would only do a few ops at a time (in the CI, or otherwise). Checking for _all_ ops will definitely take upwards of a 2-3 days at least, so I don't think we would be able to do it in CI.

Ah, I see. I had not read the comment in the other PR yet. Makes sense!

Reply:
> I was thinking if there is a way for equivalence check to enforce that different bitvector suffixes were used for the two input smt files. I tried this using a regex check of the smt files in smt_equiv.py, but it isn't straightforward. There might be a way to have LLVMToSMT output the suffix that was used in the smt file itself (as comments); then smt_equiv.py can simply confirm if the suffixes are different.

@pchaigno an update: I was able to figure out a way to do a regex pattern matching check to ensure the two input files to smt_equiv.py use different bitvector suffixes, i have [pushed](https://github.com/bpfverif/smt-equiv/commits/main/) it.  So as it stands, no changes to LLVMToSMT are needed.

Reply:
Is this PR good to go? Or should we allow `--bv-suffix-override` even without `--specific-op`? I don't see a use case for that yet.

Reply:
I agree. I think it's good to go.